### PR TITLE
Upgrade integrations SDK, go modules, and go 1.16

### PR DESCRIPTION
**Pull request by roobre**
- Upgraded `github.com/newrelic/infra-integrations-sdk` to v3.6.6
- Switched to go modules 
- Upgraded pipeline to go 1.16
- Replaced gometalinter with golangci-lint

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module github.com/newrelic/nri-kafka
+
+go 1.16
+
+require (
+	github.com/Shopify/sarama v1.27.0
+	github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc // indirect
+	github.com/eapache/go-resiliency v1.2.1-0.20190716143035-b98ce2825f72 // indirect
+	github.com/eapache/queue v1.1.1-0.20180227141424-093482f3f8ce // indirect
+	github.com/newrelic/infra-integrations-sdk v3.3.2-0.20210111170317-e4dedaa80697+incompatible
```
Comment (gsanchezgavier) [inline go.mod:10]: > +incompatible means the dependency has a semver major version of 2 or higher and is not a Go module yet (it doesn't have go.mod in its source code).

If anyone curious

**Review by gsanchezgavier — APPROVED** [review]
lgtm

**Comment by roobre** [comment]
Holding this until I perform some more e2e testing, specially with kerberos

**Review by davidbrota — APPROVED** [review]
LGTM

**Pull request merged**